### PR TITLE
fix(core): message-box, fix a11y issues

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue.
Closes SAP/fundamental-ngx/#6139

## Description

Fixed message box a11y issues

##### During Implementation
1. Visual Testing:
- [n/a] visual misalignments/updates
- [n/a] check Light/Dark/HCB/HCW themes
- [n/a] RTL/LTR - proper rendering and labeling
- [n/a] responsiveness(resize)
- [n/a] Content Density (Cozy/Compact/(Condensed))
- [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [n/a] Mouse vs. Keyboard support
- [n/a] Text Truncation
2. API and functional correctness
- [n/a] check for console logs (warnings, errors)
- [n/a] API boundary values
- [n/a] different combinations of components - free style
- [n/a] change the API values during testing
3. Documentation and Example validations
- [n/a] missing API documentation or it is not understandable
- [n/a] poor examples
- [n/a] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [X] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [n/a] Run npm run build-pack-library and test in external application
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review
Checked all of them (Inna)

### CODE REVIEW

File: libs/core/src/lib/message-box/message-box-header/message-box-header.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<header
+<div
```
Comment (InnaAtanasova) [inline libs/core/src/lib/message-box/message-box-header/message-box-header.component.html:1]: Why did you change it from header to div?

Comment (Betrozov) [reply libs/core/src/lib/message-box/message-box-header/message-box-header.component.html:1]: To solve this I tried to use the ariaLabelledBy MessageBoxService option, but it didn't work, then I try to figure out what was wrong and the difference between the message field and the accessibility dialog implementation, because it is almost the same, then I found out, that the dialog uses the div instead of header and I changed it and it started working.

how to explain this magic, unfortunately, I don't know yet

**Review by InnaAtanasova — APPROVED** [review]

**Review by Smiranin — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

**Comment by puru-hk** [comment]
Issue 1: Fixed

**Review by puru-hk — APPROVED** [review]
